python/revng/api/manager.py

```diff
@@ -116,40 +116,13 @@ def deserialize_target(self, serialized_target: str) -> Optional[Target]:
         targets = self.get_targets(step_name, container_name)
         return next((t for t in targets if t.serialize() == target), None)
 
-    def _produce_target(
-        self,
-        step_name: str,
-        target: Union[Target, List[Target]],
-        container_name: str,
-    ) -> Dict[str, str | bytes]:
-        if isinstance(target, Target):
-            targets = [target]
-        else:
-            targets = target
-
-        _step, _container = self._get_step_container_ptr(step_name, container_name)
-        product = _api.rp_manager_produce_targets(
-            self._manager, len(targets), [t._target for t in targets], _step, _container
-        )
-        if not product:
-            # TODO: we really should be able to provide a detailed error here
-            raise RevngException("Failed to produce targets")
-
-        result = {}
-        for target in targets:
-            extracted_target = target.extract()
-            if extracted_target is None:
-                raise RevngException(f"Target {target.serialize()} extraction failed")
-            result[target.serialize()] = extracted_target
-        return result
-
     def produce_target(
         self,
         step_name: str,
         target: None | str | List[str],
         container_name: Optional[str] = None,
         only_if_ready=False,
-    ) -> Dict[str, str | bytes]:
+    ) -> Dict[str, str | bytes] | Error:
```

Comment:
What's the ratio around returning `Error` as opposed to throwing an exception?

Reply:
Related to [this](https://github.com/revng/revng/pull/328#discussion_r1347297699) but basically it makes it clear at the API level that an error might be returned, so that it can't be ignored by callers and we notice it via `mypy`